blueocean-dashboard/src/main/js/components/records.jsx

```diff
@@ -23,7 +23,7 @@ export class PipelineRecord extends Record({
     weatherScore: 0,
 }) {
     isFolder() {
-        return this.numberOfFolders !== null;
+        return this._class === 'io.jenkins.blueocean.service.embedded.rest.PipelineFolderImpl';
```

Comment:
as discussed, lets use _class for now (other logic is hard to get right, surprisingly).